taichi/ir/expr.h

```diff
@@ -133,7 +133,9 @@ Expr expr_rand() {
   return taichi::lang::expr_rand(get_data_type<T>());
 }
 
-Expr snode_append(SNode *snode, const ExprGroup &indices, const Expr &val);
+Expr snode_append(SNode *snode,
```

Comment:
The semantics are little tricky here, since we won't append a `list`, but each element of it. 

Can we at least add a comment to say that this function flattens `std::vector<Expr> val` and append each element in sequence?

Reply:
The user passes in a struct, and this function allocates the space for the new struct and assigns value to every element in the struct in sequence.

Reply:
sure, can we add this as a code comment to explain how `snode_append()` works? 

Suppose if you call `snode_append()` 3 times to append a `struct1`, a `scalar`, and another `struct2` consecutively, then essentially we'll flatten + squeeze everything, then store each value to the Place snodes under a Dynamic snode. This is much different compared with the former behavior (append scalar only) and does worth a comment to clarify how it works.